cmd/ipfs/pinmfs.go

```diff
@@ -169,12 +169,16 @@ func pinMFS(
 		pinName = fmt.Sprintf("policy/%s/mfs", node.Identity().String())
 	}
 
-	// check if same pin exists
-	lsPinCh, lsErrCh := c.Ls(ctx, pinclient.PinOpts.FilterCIDs(cid), pinclient.PinOpts.FilterName(pinName))
-	pinFound := false
+	// check if MFS pin exists (across all possible states) and inspect its CID
+	pinStatuses := []pinclient.Status{pinclient.StatusQueued, pinclient.StatusPinning, pinclient.StatusPinned, pinclient.StatusFailed}
```

Comment:
Why failed? If the pinning service failed to pin, wouldn't we want it to try again?

Reply:
This is a good point!

- [ ] TODO: check status and don't skip if the CID is the same as current MFS root.